x/mongo/driver/topology/pool.go

```diff
@@ -825,12 +855,60 @@ func (p *pool) checkInNoEvent(conn *connection) error {
 	return nil
 }
 
+// clear calls clearImpl internally with a false interruptAllConnections value.
+func (p *pool) clear(err error, serviceID *primitive.ObjectID) {
+	p.clearImpl(err, serviceID, false)
+}
+
+// clearAll does same as the "clear" method but interrupts all connections.
+func (p *pool) clearAll(err error, serviceID *primitive.ObjectID) {
+	p.clearImpl(err, serviceID, true)
+}
+
+// interruptConnections interrupts the input connections.
+func (p *pool) interruptConnections(conns []*connection) {
+	for _, conn := range conns {
+		_ = conn.closeWithErr(poolClearedError{
+			err:     fmt.Errorf("interrupted"),
+			address: p.address,
+		})
+		_ = p.checkInWithCallback(conn, func() (reason, bool) {
+			if mustLogPoolMessage(p) {
+				keysAndValues := logger.KeyValues{
+					logger.KeyDriverConnectionID, conn.driverConnectionID,
+				}
+
+				logPoolMessage(p, logger.ConnectionCheckedIn, keysAndValues...)
+			}
+
+			if p.monitor != nil {
+				p.monitor.Event(&event.PoolEvent{
+					Type:         event.ConnectionCheckedIn,
```

Comment:
I don't think the spec requires that check-in logs/events are emitted when closing in-use connections. If we stop using check-in to close connections (recommended above), we should also not emit these logs/events.